test/circuit_optimizer/test_collect_blocks.cpp

```diff
@@ -37,20 +53,63 @@ TEST(CollectBlocks, collectMultipleSingleQubitGates) {
   qc.x(0);
   qc.x(1);
   std::cout << qc << "\n";
-  qc::CircuitOptimizer::collectBlocks(qc, 1);
+  qc::CircuitOptimizer::collectBlocks(qc, 1, false);
   std::cout << qc << "\n";
   EXPECT_EQ(qc.size(), 2);
   EXPECT_TRUE(qc.front()->isCompoundOperation());
   EXPECT_TRUE(qc.back()->isCompoundOperation());
 }
 
+TEST(CollectBlocks, collectMultipleSingleQubitCliffordGates) {
+  QuantumComputation qc(2);
+  qc.h(0);
+  qc.h(1);
+  qc.t(0);
+  qc.t(1);
+  qc.x(0);
+  qc.x(1);
+  std::cout << qc << "\n";
+  qc::CircuitOptimizer::collectBlocks(qc, 1, true);
+  std::cout << qc << "\n";
+  EXPECT_EQ(qc.size(), 6);
+}
+
+TEST(CollectBlocks, collectMultipleCliffordGates) {
+  QuantumComputation qc(2);
+  qc.h(0);
+  qc.h(1);
+  qc.t(0);
+  qc.x(0);
+  qc.x(1);
+  std::cout << qc << "\n";
+  qc::CircuitOptimizer::collectBlocks(qc, 2, true);
+  std::cout << qc << "\n";
+  EXPECT_EQ(qc.size(), 4);
+  EXPECT_TRUE(qc.front()->isCompoundOperation());
+}
```

Comment:
This test does not seem quite right.
Looking at the circuit, I would have imagined there to be 3 blocks in the resulting circuit (H(0), H(1), X(1)), (T(0)), and (X(0)).
Please check the implementation again. and devise reasonable tests.
Instead of simply asserting the number of operations, assert what the actual operations are to make sure that the pass is doing what it is supposed to do.